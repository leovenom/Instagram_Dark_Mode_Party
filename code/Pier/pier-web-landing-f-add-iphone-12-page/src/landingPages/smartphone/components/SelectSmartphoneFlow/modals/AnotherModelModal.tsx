import { useState } from "react";
import * as React from "react";
import styled from "styled-components";
import { useFormik } from "formik";
import * as Yup from "yup";

import validateCPF from "helpers/validateCPF";
import config from "config";
import Modal from "ui/Modal";
import Text from "ui/Text";
import Input from "ui/Input";
import Button from "ui/Button";
import { useToast } from "ui/Toast";

import breakpoints from "ui/theme/breakpoints";
import jsonToQueryString from "helpers/jsonToQueryString";

interface Props {
  isOpen: boolean;
  hide: () => void;
  toggleAnotherModelSuccessSubmition: () => void;
}

const StyledModal = styled(Modal)`
  section {
    box-sizing: border-box;
    overflow-y: auto;

    form {
      margin-top: 24px;
    }
  }

  ${breakpoints.tablet} {
    max-width: 700px;
    margin: 10px auto;
  }
`;

const InputContainer = styled.div`
  padding-top: 24px;
`;

const modelValidationSchema = Yup.object({
  full_name: Yup.string()
    .required("Nome obrigat칩rio")
    .min(3, "Nome muito curto"),
  email: Yup.string().email("Email inv치lido").required("Email obrigat칩rio"),
  cpf: Yup.string().test("is-cpf", "CPF inv치lido", (value) =>
    validateCPF(value)
  ),
  manufacturer: Yup.string().required("Fabricante obrigat칩rio"),
  model: Yup.string().required("Modelo obrigat칩rio"),
});

const modelInitialValues = {
  full_name: "",
  email: "",
  cpf: "",
  manufacturer: "",
  model: "",
  kind: "android_prospect",
};

const AnotherModelModal: React.FC<Props> = ({
  isOpen,
  hide,
  toggleAnotherModelSuccessSubmition,
}) => {
  const toast = useToast();
  const [isLoading, setIsLoading] = useState(false);

  const {
    handleSubmit,
    handleChange,
    values,
    errors,
    setFieldTouched,
    touched,
    isValid,
  } = useFormik({
    initialValues: modelInitialValues,
    validationSchema: modelValidationSchema,
    validateOnMount: true,
    onSubmit: async (values) => {
      const queryString = jsonToQueryString(values);
      let result;
      setIsLoading(true);
      try {
        result = await fetch(`${config.PIER_SITE_URL}/faqs${queryString}`, {
          method: "POST",
          headers: {
            "Content-type": "application/x-www-form-urlencoded; charset=UTF-8",
          },
        });

        if (result.status !== 200) {
          toast.error(
            `游땩 Ocorreu um erro.
            Tente novamente ou entre em contato com o suporte!`
          );
          return;
        }

        hide();
        toggleAnotherModelSuccessSubmition();
      } catch {
        toast.error(
          `Oops!! Algo deu errado! 游땩 Tente novamente ou entre em contato com o suporte.`
        );
      } finally {
        setIsLoading(false);
      }
    },
  });

  return (
    <StyledModal
      title="N칚o encontrou seu modelo?"
      isOpen={isOpen}
      hide={hide}
      contentLabel="Modal para submiss칚o de novos modelos"
    >
      <section>
        <Text variant="bodySmall">
          No momento protegemos apenas os modelos listados. Nosso desejo 칠
          acolher cada vez mais modelos em nossa comunidade. Deixe aqui seus
          dados e te informaremos assim que tivermos novidades!
        </Text>

        <form translate="yes" onSubmit={handleSubmit}>
          <InputContainer>
            <Input
              name="full_name"
              type="text"
              label="Nome completo"
              placeholder="Carlos Toledo"
              onChange={handleChange}
              value={values["full_name"]}
              error={touched.full_name ? errors.full_name : ""}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="email"
              type="text"
              label="Email"
              placeholder="meu@email.com"
              onChange={handleChange}
              value={values.email}
              error={touched.email ? errors.email : ""}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="cpf"
              type="text"
              label="CPF"
              mask="cpf"
              placeholder="123.456.789-00"
              onChange={handleChange}
              value={values.cpf}
              error={touched.cpf ? errors.cpf : ""}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="manufacturer"
              type="text"
              label="Fabricante"
              placeholder="Motorola"
              onChange={handleChange}
              value={values.manufacturer}
              error={touched.manufacturer ? errors.manufacturer : ""}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="model"
              type="text"
              label="Modelo"
              placeholder="G8"
              onChange={handleChange}
              value={values.model}
              error={touched.model ? errors.model : ""}
            />
          </InputContainer>
          <Button
            mt={24}
            fullWidth
            type="submit"
            disabled={isLoading || !isValid}
            isLoading={isLoading}
          >
            Enviar
          </Button>
        </form>
      </section>
    </StyledModal>
  );
};

export default AnotherModelModal;
