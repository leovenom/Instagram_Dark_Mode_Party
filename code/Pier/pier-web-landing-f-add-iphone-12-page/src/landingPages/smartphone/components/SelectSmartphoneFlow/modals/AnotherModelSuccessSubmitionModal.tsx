import * as React from "react";

import Modal from "ui/Modal";
import Text from "ui/Text";

interface Props {
  isOpen: boolean;
  hide: () => void;
}

const AnotherModelSuccessSubmitionModal: React.FC<Props> = ({
  isOpen,
  hide,
}) => {
  return (
    <Modal
      title="Interesse: Cobertura para novos dispositivos"
      isOpen={isOpen}
      hide={hide}
      contentLabel="Exemplo de Modal básico que preenche toda a tela"
    >
      <Text variant="sectionSecondary" bold>
        Obrigado!
      </Text>
      <Text variant="bodyLarge">
        Ficamos felizes pelo seu interesse em participar de nossa comunidade.
      </Text>
      <Text variant="bodyLarge">
        Agora é só aguardar que em breve teremos mais informações ;)
      </Text>
    </Modal>
  );
};

export default AnotherModelSuccessSubmitionModal;
