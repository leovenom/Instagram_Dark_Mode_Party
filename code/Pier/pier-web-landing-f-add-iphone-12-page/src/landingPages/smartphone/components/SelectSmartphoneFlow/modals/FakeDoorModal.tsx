import { useState, useEffect } from "react";
import * as React from "react";
import styled from "styled-components";
import { useFormik } from "formik";
import ImageResponsive from "ui/ImageResponsive";

import config from "config";
import Modal from "ui/Modal";
import Text from "ui/Text";
import Input from "ui/Input";
import Button from "ui/Button";
import { useToast } from "ui/Toast";
import { smartphoneTracker } from "helpers/mixpanelTracker";

import breakpoints from "ui/theme/breakpoints";
import jsonToQueryString from "helpers/jsonToQueryString";

import data from "../data";
import spacings from "ui/theme/spacings";

const StyledModal = styled(Modal)`
  section {
    box-sizing: border-box;
    overflow-y: auto;
    form {
      margin-top: 24px;
    }
  }
  ${breakpoints.tablet} {
    max-width: 700px;
    margin: 10px auto;
  }
`;

const InputContainer = styled.div`
  padding-top: 24px;
`;

interface Props {
  isOpen: boolean;
  hide: () => void;
  model: { manufacturer: string; model: string };
  toggleFakeDoorSuccessSubmitionModal: () => void;
}

const FakeDoorModal: React.FC<Props> = ({
  isOpen,
  hide,
  model: { manufacturer, model } = {},
  toggleFakeDoorSuccessSubmitionModal,
}) => {
  const initialFormData = {
    manufacturer,
    model,
    email: "",
    phone: "",
  };

  const toast = useToast();

  const [isLoading, setIsLoading] = useState(false);

  const onFormSubmit = async (values: typeof initialFormData) => {
    setIsLoading(true);

    smartphoneTracker.trackButton(data.fakeDoor.form.tracking.buttonCliked, {
      brand: values.manufacturer,
      model: values.model,
      whatsapp_check: !!values.phone,
    });

    try {
      const result = await fetch(
        `${config.PIER_SITE_URL}/faqs${jsonToQueryString(
          values
        )}&kind=fakedoor`,
        {
          method: "POST",
          headers: {
            "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
          },
        }
      );

      if (result.status !== 200) throw new Error();

      hide();
      toggleFakeDoorSuccessSubmitionModal();
    } catch {
      toast.error(data.fakeDoor.form.error);
    } finally {
      setIsLoading(false);
    }
  };

  const {
    handleSubmit,
    handleChange,
    handleBlur,
    values,
    errors,
    touched,
    isValid,
  } = useFormik({
    enableReinitialize: true,
    initialValues: initialFormData,
    validationSchema: data.fakeDoor.form.schema,
    validateOnMount: true,
    isInitialValid: false,
    onSubmit: onFormSubmit,
  });

  useEffect(() => {
    if (isOpen)
      smartphoneTracker.trackModal(data.fakeDoor.form.tracking.screenViewed);
  }, [data, isOpen]);

  return (
    <StyledModal
      mobileFullScreen
      title={data.fakeDoor.form.title}
      isOpen={isOpen}
      hide={hide}
      contentLabel={data.fakeDoor.form.contentLabel}
    >
      <section>
        <ImageResponsive
          src={data.fakeDoor.form.image.src}
          alt={data.fakeDoor.form.image.alt}
        />

        <Text mt={spacings.TERA} mb={spacings.MEGA} variant="bodySmall">
          {data.fakeDoor.form.description}
        </Text>

        <form translate="yes" onSubmit={handleSubmit}>
          <InputContainer>
            <Input
              name="email"
              type="text"
              label={data.fakeDoor.form.inputs.email.label}
              placeholder={data.fakeDoor.form.inputs.email.placeholder}
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.email}
              error={touched.email && errors.email}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="manufacturer"
              type="text"
              label={data.fakeDoor.form.inputs.manufacturer.label}
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.manufacturer}
              error={touched.manufacturer && errors.manufacturer}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="model"
              type="text"
              label={data.fakeDoor.form.inputs.model.label}
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.model}
              error={touched.model && errors.model}
            />
          </InputContainer>
          <InputContainer>
            <Input
              name="phone"
              type="text"
              label={data.fakeDoor.form.inputs.phone.label}
              placeholder={data.fakeDoor.form.inputs.phone.placeholder}
              onChange={handleChange}
              value={values.phone}
              mask="phoneNumber"
              required={false}
              helpField={data.fakeDoor.form.inputs.phone.disclaimer}
              maskPlaceholder="_"
            />
          </InputContainer>
          <Button
            mt={spacings.GIGA}
            fullWidth
            type="submit"
            disabled={isLoading || !isValid}
            isLoading={isLoading}
            variant="secondary"
          >
            {data.fakeDoor.form.submit}
          </Button>
        </form>
      </section>
    </StyledModal>
  );
};

export default FakeDoorModal;
