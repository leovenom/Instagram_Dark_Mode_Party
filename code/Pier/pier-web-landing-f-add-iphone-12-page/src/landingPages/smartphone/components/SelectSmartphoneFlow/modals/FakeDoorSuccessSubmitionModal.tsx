import * as React from "react";
import styled from "styled-components";

import Modal from "ui/Modal";
import Text from "ui/Text";
import Button from "ui/Button";
import ImageResponsive from "ui/ImageResponsive";
import spacings from "ui/theme/spacings";

import data from "../data";

const ContentContainer = styled.div`
  display: flex;
  padding-top: calc(50% - 65px);
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

interface Props {
  isOpen: boolean;
  hide: () => void;
}

const FakeDoorSuccessSubmitionModal: React.FC<Props> = ({ isOpen, hide }) => {
  return (
    <Modal
      mobileFullScreen
      title={data.fakeDoor.success.title}
      isOpen={isOpen}
      hide={hide}
      contentLabel={data.fakeDoor.success.contentLabel}
    >
      <ContentContainer>
        <ImageResponsive
          src={data.fakeDoor.success.image.src}
          alt={data.fakeDoor.success.image.alt}
        />
        <Text align="center" mt={spacings.MEGA} variant="bodySmall">
          {data.fakeDoor.success.description}
        </Text>
        <Button
          mt={spacings.TERA + spacings.KILO}
          fullWidth
          onClick={hide}
          variant="secondary"
        >
          {data.fakeDoor.success.button}
        </Button>
      </ContentContainer>
    </Modal>
  );
};

export default FakeDoorSuccessSubmitionModal;
