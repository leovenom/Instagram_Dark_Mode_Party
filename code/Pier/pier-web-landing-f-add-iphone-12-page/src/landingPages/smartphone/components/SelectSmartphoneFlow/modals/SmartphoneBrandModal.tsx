import * as React from "react";

import { ListItem, ExtraOptions } from "../components";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import { MODAL_STEPS } from "helpers/constants";
import { ModelsContainer } from "../styles";
import { filterModelsByFakeDoor } from "../utils/functions";

interface Props {
  smartphoneModels: any;
  currentPhoneState: any;
  modalKey: string;
  setPhone: (value) => void;
  hideModal: () => void;
  toggleAnotherModel: () => void;
  withFakeDoor?: boolean;
  eventIdentifier?: string;
}

const SmartphoneBrandModal: React.FC<Props> = ({
  smartphoneModels,
  currentPhoneState,
  modalKey,
  setPhone,
  hideModal,
  toggleAnotherModel,
  withFakeDoor,
  eventIdentifier = "",
}) => {
  React.useEffect(() => {
    smartphoneTracker.trackScreen(`${eventIdentifier} ChooseBrand`);
  }, []);

  const handleSelectBrand = (name, deviceOs, manufacturer) => {
    smartphoneTracker.track(`${eventIdentifier} ChooseBrand Item Selected`, {
      brand: name,
    });
    setPhone({
      ...currentPhoneState,
      serie: name,
      deviceOs,
      manufacturer,
    });
  };

  const filteredModels = (): any =>
    filterModelsByFakeDoor({
      models: smartphoneModels,
      withFakeDoor,
      dataType: "isFakeDoor",
    });

  return (
    <ModelsContainer data-testid="smartphone-brand-modal" key={modalKey}>
      <>
        {filteredModels().map(({ name, deviceOs, manufacturer }) => (
          <ListItem
            key={name}
            onClick={() => handleSelectBrand(name, deviceOs, manufacturer)}
          >
            {name}
          </ListItem>
        ))}
        <ExtraOptions
          extraOptionType={MODAL_STEPS.MODELS}
          hideModal={hideModal}
          toggleAnotherModel={toggleAnotherModel}
        />
      </>
    </ModelsContainer>
  );
};

export default SmartphoneBrandModal;
