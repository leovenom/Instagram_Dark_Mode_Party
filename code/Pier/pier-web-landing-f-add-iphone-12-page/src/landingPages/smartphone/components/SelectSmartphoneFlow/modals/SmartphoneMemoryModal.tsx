import * as React from "react";

import { ListItem, ExtraOptions } from "../components";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import { setSmartphoneMemory } from "../utils/functions";
import { MODAL_STEPS } from "helpers/constants";

import { ModelsContainer } from "../styles";

interface Props {
  currentPhoneState: any;
  modalKey: string;
  setPhone: (value) => void;
  hideModal: () => void;
  toggleAnotherModel: () => void;
  eventIdentifier?: string;
}

const SmartphoneMemoryModal: React.FC<Props> = ({
  currentPhoneState,
  hideModal,
  toggleAnotherModel,
  setPhone,
  modalKey,
  eventIdentifier = "",
}) => {
  React.useEffect(() => {
    smartphoneTracker.trackScreen(`${eventIdentifier} ChooseModelCapacity`);
  }, []);

  const handleSelectMemory = (memory) => {
    smartphoneTracker.track(
      `${eventIdentifier} ChooseModelCapacity Item Selected`,
      {
        capacity: memory,
      }
    );
    setSmartphoneMemory({ memory, setPhone, currentPhoneState });
  };

  return (
    <ModelsContainer
      data-testid="smartphone-memory-modal"
      animate
      key={modalKey}
    >
      {currentPhoneState.model?.memories?.map((memory) => (
        <ListItem key={memory} onClick={() => handleSelectMemory(memory)}>
          {memory}
        </ListItem>
      ))}
      <ExtraOptions
        extraOptionType={MODAL_STEPS.MEMORY}
        hideModal={hideModal}
        toggleAnotherModel={toggleAnotherModel}
      />
    </ModelsContainer>
  );
};

export default SmartphoneMemoryModal;
