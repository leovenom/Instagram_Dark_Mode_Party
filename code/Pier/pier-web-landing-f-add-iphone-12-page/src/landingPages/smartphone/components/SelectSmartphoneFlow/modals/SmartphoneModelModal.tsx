import * as React from "react";

import { ExtraOptions, ListItem } from "../components";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import { MODAL_STEPS } from "helpers/constants";
import { filterModelsByFakeDoor, setSmartphoneModel } from "../utils/functions";
import { ModelsContainer } from "../styles";

interface Props {
  smartphoneModels: any;
  currentPhoneState: any;
  modalKey: string;
  setPhone: (value) => void;
  toggleFakeDoor: () => void;
  onModelSelectByFakeDoor: (value) => void;
  hideModal: () => void;
  toggleAnotherModel: () => void;
  withFakeDoor?: boolean;
  eventIdentifier?: string;
}

const SmartphoneModelModal: React.FC<Props> = ({
  smartphoneModels,
  currentPhoneState,
  modalKey,
  setPhone,
  toggleFakeDoor,
  onModelSelectByFakeDoor,
  hideModal,
  toggleAnotherModel,
  withFakeDoor,
  eventIdentifier = "",
}) => {
  React.useEffect(() => {
    smartphoneTracker.trackScreen(`${eventIdentifier} ChooseModel`);
  }, []);

  const getModelsOfTheSelectedLine = (): any =>
    currentPhoneState.serie
      ? smartphoneModels.find(({ name }) => name === currentPhoneState.serie)
      : [];

  const filteredModels = (): any =>
    filterModelsByFakeDoor({
      models: getModelsOfTheSelectedLine()?.models,
      withFakeDoor,
      dataType: "memories",
    });

  const handleSelectSmartphoneModel = (model) => {
    smartphoneTracker.track(`${eventIdentifier} ChooseModel Item Selected`, {
      model: model.name,
    });

    const isModelPlanAvailable = !!model.memories;
    if (isModelPlanAvailable) {
      return setSmartphoneModel({ model, currentPhoneState, setPhone });
    }

    onModelSelectByFakeDoor({
      manufacturer: currentPhoneState.manufacturer,
      model: model.name,
    });

    hideModal();
    toggleFakeDoor();
  };

  return (
    <ModelsContainer
      data-testid="smartphone-model-modal"
      animate
      key={modalKey}
    >
      {filteredModels().map((model) => (
        <ListItem
          key={model.name}
          onClick={() => handleSelectSmartphoneModel(model)}
        >
          {model.name}
        </ListItem>
      ))}
      <ExtraOptions
        extraOptionType={MODAL_STEPS.SERIES}
        hideModal={hideModal}
        toggleAnotherModel={toggleAnotherModel}
      />
    </ModelsContainer>
  );
};

export default SmartphoneModelModal;
