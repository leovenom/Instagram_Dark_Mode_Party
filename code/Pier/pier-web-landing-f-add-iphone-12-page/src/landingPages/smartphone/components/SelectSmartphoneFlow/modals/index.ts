import SmartphoneBrandModal from "./SmartphoneBrandModal";
import SmartphoneMemoryModal from "./SmartphoneMemoryModal";
import SmartphoneModelModal from "./SmartphoneModelModal";
import FakeDoorModal from "./FakeDoorModal";
import AnotherModelModal from "./AnotherModelModal";
import FakeDoorSuccessSubmitionModal from "./FakeDoorSuccessSubmitionModal";
import AnotherModelSuccessSubmitionModal from "./FakeDoorSuccessSubmitionModal";

export {
  SmartphoneBrandModal,
  SmartphoneMemoryModal,
  SmartphoneModelModal,
  AnotherModelModal,
  FakeDoorModal,
  FakeDoorSuccessSubmitionModal,
  AnotherModelSuccessSubmitionModal,
};
