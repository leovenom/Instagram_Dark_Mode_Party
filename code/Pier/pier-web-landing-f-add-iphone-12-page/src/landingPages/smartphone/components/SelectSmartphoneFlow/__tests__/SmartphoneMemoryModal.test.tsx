import { SmartphoneMemoryModal } from "../modals";
import { smartphoneTracker } from "helpers/mixpanelTracker";
import smartphoneModels from "helpers/smartphoneModels";
import { renderWithTheme } from "helpers/testUtils";
import { fireEvent } from "@testing-library/react";

jest.mock("helpers/mixpanelTracker");

const runPropsSetup = ({
  currentPhoneState,
  withFakeDoor = true,
  eventIdentifier = "",
}) => ({
  smartphoneModels,
  currentPhoneState,
  modalKey: "one",
  setPhone: () => {},
  hideModal: () => {},
  toggleAnotherModel: () => {},
  withFakeDoor,
  eventIdentifier,
});

const getIphone6Memories = (): any => {
  const getAppleModels = (): any =>
    smartphoneModels.find((serie) => serie.name === "Apple") ?? [];

  return (
    getAppleModels().models.find((model) => model.name === "iPhone 6S") ?? []
  );
};

const renderComponentByProps = (props) =>
  renderWithTheme(<SmartphoneMemoryModal {...props} />);

describe("<SmartphoneMemoryModal />", () => {
  it("Should render without errors", () => {
    const propsSetup = runPropsSetup({
      currentPhoneState: {
        serie: "Apple",
        deviceOs: "ios",
        model: { name: "iPhone 6S" },
        memory: null,
        manufacturer: null,
      },
    });

    const { container } = renderComponentByProps(propsSetup);
    expect(container).toMatchSnapshot();
  });

  it("Should render the <SmartphoneMemoryModal /> with all memories of selected model", () => {
    const propsSetup = runPropsSetup({
      currentPhoneState: {
        serie: "Apple",
        deviceOs: "ios",
        model: { name: "iPhone 6S", memories: getIphone6Memories().memories },
        memory: null,
        manufacturer: null,
      },
    });
    const { getByText } = renderComponentByProps(propsSetup);

    getIphone6Memories().memories.forEach((memory) => {
      getByText(memory);
    });
  });

  describe("When click at an item", () => {
    it("Should trigger a mixpanel event with default event", () => {
      const propsSetup = runPropsSetup({
        withFakeDoor: false,
        currentPhoneState: {
          serie: "Apple",
          deviceOs: "ios",
          model: { name: "iPhone 6S", memories: getIphone6Memories().memories },
          memory: null,
          manufacturer: null,
        },
      });

      const { getByText } = renderComponentByProps(propsSetup);

      fireEvent.click(getByText("64GB"));

      expect(smartphoneTracker.track).toHaveBeenCalledWith(
        " ChooseModelCapacity Item Selected",
        {
          capacity: "64GB",
        }
      );
    });

    it("Should trigger a mixpanel event with a custom event", () => {
      const propsSetup = runPropsSetup({
        withFakeDoor: false,
        currentPhoneState: {
          serie: "Apple",
          deviceOs: "ios",
          model: { name: "iPhone 6S", memories: getIphone6Memories().memories },
          memory: null,
          manufacturer: null,
        },
        eventIdentifier: "Tabela Pipe",
      });
      const { getByText } = renderComponentByProps(propsSetup);

      fireEvent.click(getByText("64GB"));

      expect(smartphoneTracker.track).toHaveBeenCalledWith(
        "Tabela Pipe ChooseModelCapacity Item Selected",
        {
          capacity: "64GB",
        }
      );
    });
  });
});
