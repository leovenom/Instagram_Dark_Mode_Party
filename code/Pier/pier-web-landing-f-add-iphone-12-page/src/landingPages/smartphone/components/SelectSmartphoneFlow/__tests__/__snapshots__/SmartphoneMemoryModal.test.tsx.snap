// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmartphoneMemoryModal /> Should render without errors 1`] = `
<div>
  <ul
    class="styles__ModelsContainer-sc-1ipelzl-1 jiJlpd"
    data-testid="smartphone-memory-modal"
  >
    <li
      class="styles__ItemContainer-sc-1ipelzl-2 gxJPSZ"
      role="button"
      tabindex="0"
    >
      NÃ£o Sei
      <svg
        fill="#24263B"
        height="14"
        viewBox="0 0 10 14"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.976 7.531a.736.736 0 000-1.031L2.913.406c-.312-.281-.781-.281-1.062 0l-.72.719c-.28.281-.28.75 0 1.063L5.945 7l-4.812 4.844c-.281.312-.281.781 0 1.062l.719.719c.28.281.75.281 1.062 0l6.063-6.094z"
        />
      </svg>
    </li>
  </ul>
</div>
`;
